fix: It is still possible to mint two identic images

hassnian opened the pull request:
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring


## Context

> using auto-teleport instead of normal mint

@prury you shouldn't have to use auto-teleport to replicate this, the page refresh is done after you close the success modal, opening the listing modal and then closing it allows you to skip it.

> it would be interesting to refactor/change the code in a way that this could be handled better.

#10148 fixes that

- [x] Closes #10312
- [x] Closes #10335

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)


## Screenshot 📸

- [x] My fix has changed **something** on UI; 


https://github.com/kodadot/nft-gallery/assets/44554284/aea84e51-5375-42e4-9076-6255b1377c63

prury commented:
> 
> @prury you shouldn't have to use auto-teleport to replicate this, the page refresh is done after you close the success modal, opening the listing modal and then closing it allows you to skip it.

i see, ty.

but i think the problem still occurs, i was able to close the listing modal in this PR as well without page refresh when closing.

hassnian commented:
@prury  

## issue 1: page state not resetting

I have deleted the page refresh , let's see if the page app resets properly after a mint

pls test back to back mints using the normal  flow and also with auto-teleport and combine 

 ## issue 2: user's mint count not being updated after mint

now after a mint it moves to next range, this generates a new hash 

as you can see now you can mint multiple times and it's not giving you the same nft 

https://github.com/kodadot/nft-gallery/assets/44554284/7ba7faef-3f24-47a6-a731-01379b399f6d

## issue 3: modal state is being reset before the animation ends

I've also fixed https://github.com/kodadot/nft-gallery/issues/10335

https://github.com/kodadot/nft-gallery/assets/44554284/3cc053c5-69c4-4713-bfde-15ab869651d7

prury commented:
> issue 1: page state not resetting after a mint without a page reset
> 
> I have deleted the page refresh , let's see if the app resets properly after a mint
> 
> pls test back to back mints using the normal flow and also with auto-teleport and combine
> ## issue 2: user's mint count not being updated after mint
> 
> now after a mint it moves to next entropy range, this generates a new hash , (00:45)
> 

nice, i could observe that the count was properly updated and a new hash is being generated as well in both auto teleport and normal mint without having to do a page refresh. 

> ## issue 3: modal state is being reset before the animation ends
> 
> #10335

nice, thank you for solving this one as well

everything that i tested so far worked, but i want to revisit this PR and do more tests, will wait for dev reviews.

hassnian commented on components/collection/drop/PaidGenerative.vue:
```diff
@@ -116,7 +116,7 @@ const openMintModal = () => {
 
 const handleMintModalClose = () => {
   closeMintModal()
-  clearMassMint()
+  setTimeout(clearMassMint, NEO_MODAL_ANIMATION_DURATION)
 }
```
here we are not listening for `isModalActive`, this is reacting to the @close event , when going through the auto-teleport flow we set `isModalActive` to `false`, but in that case we don't want to clear

preschian reviewed (approved):
since we are changing how the user [get hash](https://github.com/kodadot/ops-internal/issues/1659), it solve this issue also

prury commented:
> since we are changing how the user [get hash](https://github.com/kodadot/ops-internal/issues/1659), it solve this issue also

do you think we should close this PR then?

preschian commented:
> do you think we should close this PR then?

let's merge this one
